src/host/vsts.ts

```diff
@@ -0,0 +1,40 @@
+import Host from './host';
+import GitInfo from '../gitInfo'
+import ConfigInfo from "../configInfo";
+
+export default class Vsts implements Host {
+    /**
+     * The regular expression to match the VSTS Git URL.
+     * @example https://my-tenant.visualstudio.com/DefaultCollection/MyCollection/_git/my-repo
+     * @example ssh://my-tenant@my-tenant.visualstudio.com:22/DefaultCollection/MyCollection/_git/my-repo
+     */
+    private static urlRegex: RegExp = /(?:https:\/\/|ssh:\/\/.+@)(\w+)\.visualstudio\.com(?:\:\d+)?\/([/\w]+)\/_git\/([^/]+)/i;
```

Comment:
`\/([/\w]+)\/` do we need to consider the case such like  ssh://my-tenant@my-tenant.visualstudio.com:22/**Default.Collection**/MyCollection/_git/my-repo